app/data_model/progress_store.py

```diff
@@ -15,96 +16,120 @@ class ProgressStore:
     that have been started.
     """
 
-    def __init__(self, in_progress_sections: Mapping = None) -> None:
+    def __init__(self, in_progress_sections: List[Dict] = None) -> None:
         """
         Instantiate a ProgressStore object that tracks the status of sections and its completed locations
         Args:
             in_progress_sections: A hierarchical dict containing the section status and completed locations
         """
         self._is_dirty = False  # type: bool
-        self._progress = {}  # type: MutableMapping
+        self._progress = self._build_map(in_progress_sections or [])  # type: Dict
 
-        if not in_progress_sections:
-            return
+    def __contains__(self, section_key) -> bool:
+        return section_key in self._progress
 
-        for section_id, serialised_section in in_progress_sections.items():
-            self._progress[section_id] = {
-                'status': serialised_section['status'],
-                'locations': [
-                    Location.from_dict(location_dict=location)
-                    for location in serialised_section['locations']
-                ],
-            }
-
-    def __contains__(self, section_id):
-        return section_id in self._progress
+    @staticmethod
+    def _build_map(in_progress_sections: List[Dict]) -> Dict:
+        """ Builds the progress_store's data structure from a list of progress dictionaries"""
+        return {
+            (section['section_id'], section.get('list_item_id')): Progress.from_dict(
+                section
+            )
+            for section in in_progress_sections
+        }
 
     @property
     def is_dirty(self) -> bool:
         return self._is_dirty
 
     @property
-    def completed_section_ids(self) -> List[str]:
-        complete_section_ids = []
-        for section_id, section_progress in self._progress.items():
-            if section_progress['status'] == CompletionStatus.COMPLETED:
-                complete_section_ids.append(section_id)
-
-        return complete_section_ids
-
-    def update_section_status(self, section_id: str, section_status: str) -> None:
-        if section_id in self._progress:
-            self._progress[section_id]['status'] = section_status
+    def completed_section_keys(self) -> List[Tuple[str]]:
+        return [
+            section_key
+            for section_key, section_progress in self._progress.items()
+            if section_progress.status == CompletionStatus.COMPLETED
+        ]
+
+    def update_section_status(
+        self, section_status: str, section_id, list_item_id=None
+    ) -> None:
+
+        section_key = (section_id, list_item_id)
+        if section_key in self._progress:
```

Comment:
To be honest, I personally prefer LBYL but I think we should make a decision as a team on an approach because the `progress_store` went through peer reviews and was approved with the LBYL approach. 

I think changing this to EAFP is out of scope unless the team decides that is what we should go with from now on.